extensions/vscode/src/multiStepInputs/newConnectCloudCredential.ts

```diff
@@ -0,0 +1,537 @@
+  const productName: ProductName = ProductName.CONNECT_CLOUD;
+
+  const connectCloudData: ConnectCloudData = {
+    accounts: [],
+    auth: {
+      deviceCode: "",
+      userCode: "",
+      verificationURI: "",
+      interval: 0,
+    },
+  };
+
+  enum step {
+    INIT_DEVICE_AUTH = "initDeviceAuth",
+    AUTHENTICATE = "authenticate",
+    RETRIEVE_ACCOUNTS = "retrieveAccounts",
+    DETERMINE_ACCT_FLOW = "determineAccountFlow",
+    INPUT_ACCT = "inputAccount",
+    INPUT_SIGNUP = "inputSignup",
+    INPUT_CRED_NAME = "inputCredentialName",
+  }
+
+  const steps: Record<
+    step,
+    (input: MultiStepInput, state: MultiStepState) => Promise<void | InputStep>
+  > = {
+    [step.INIT_DEVICE_AUTH]: initDeviceAuth,
+    [step.AUTHENTICATE]: authenticate,
+    [step.RETRIEVE_ACCOUNTS]: retrieveAccounts,
+    [step.DETERMINE_ACCT_FLOW]: determineAccountFlow,
+    [step.INPUT_ACCT]: inputAccount,
+    [step.INPUT_SIGNUP]: inputSignup,
+    [step.INPUT_CRED_NAME]: inputCredentialName,
+  };
+
+  // update the device auth data for Connect Cloud
+  const updateConnectCloudAuthData = (data?: DeviceAuth) => {
+    connectCloudData.auth.deviceCode = data?.deviceCode || "";
+    connectCloudData.auth.verificationURI = data?.verificationURI || "";
+    connectCloudData.auth.userCode = data?.userCode || "";
+    connectCloudData.auth.interval = data?.interval || 0;
+  };
+
+  // ***************************************************************
+  // Order of all steps for creating a new Connect Cloud credential
+  // ***************************************************************
+
+  // Initialize the device authentication
+  // Poll the device authentication
+  // Retrive the user's accounts
+  // Determine the correct next step:
+  //  - If there is only one publishable account:
+  //    Get the credential name
+  //  - If there are multiple publishable accounts:
+  //    Get selected account from account list
+  //    Get the credential name
+  //  - If there are no publishable accounts, but there is at least one account:
+  //    Get sign up for individual plan
+  //    Initialize the device authentication
+  //    Poll the device authentication
+  //    Poll for the user's new account
+  //    Determine the correct next step
+  //     - There will be only one publishable account:
+  //       Get the credential name
+  //  - If there are zero accounts for the user:
+  //    Poll for the user's new account
+  //    Determine the correct next step
+  //     - There will be only one publishable account:
+  //       Get the credential name
+  // result in calling credential API
+
+  // ***************************************************************
+  // Method which kicks off the multi-step.
+  // Initialize the state data
+  // Display the first input panel
+  // ***************************************************************
+  async function collectInputs() {
+    const state: MultiStepState = {
```

Comment:
Seems a lot of default/initialized objects are being created here, it might be good to have factories? WDYT of having methods so the steps here are a bit more tidy. e,g: `NewMultiStepState()`, `NewDeviceAuth()`, `NewAuthToken()` etc